Sources/Jobs/Scheduler/Schedule.swift

```diff
@@ -106,54 +149,100 @@ public struct Schedule: Sendable {
+        .init(minute: .specific(minute), hour: .specific(hour), date: .specific(date), month: .selection(Deque(months.sorted())), timeZone: timeZone)
     }
 
     ///  Return a schedule that generates a Date once a year
     /// - Parameters:
     ///   - month: Month on which it should return Date at
     ///   - date: Date on which it should return Date at
     ///   - hour: Hour value is should return Date at
+    ///   - minute: Minute value it should return a Date at
     ///   - timeZone: Time zone to use when scheduling
-    public static func yearly(month: Month, date: Int, hour: Int = 0, timeZone: TimeZone = .current) -> Self {
-        .init(hour: .specific(hour), date: .specific(date), month: .specific(month), timeZone: timeZone)
+    public static func yearly(month: Month, date: Int, hour: Int = 0, minute: Int = 0, timeZone: TimeZone = .current) -> Self {
+        .init(minute: .specific(minute), hour: .specific(hour), date: .specific(date), month: .specific(month), timeZone: timeZone)
     }
 
     ///  Return next date in schedule after the supplied Date
     /// - Parameter date: start date
-    public func nextDate(after date: Date = .now) -> Date? {
+    public mutating func nextDate(after date: Date) -> Date? {
         var dateComponents = DateComponents()
         dateComponents.nanosecond = 0
-        if case .specific(let second) = self.second {
-            dateComponents.second = second
-        }
-        if case .specific(let minute) = self.minute {
-            dateComponents.minute = minute
-        }
-        if case .specific(let hour) = self.hour {
-            dateComponents.hour = hour
-        }
-        if case .specific(let day) = self.day {
-            dateComponents.weekday = day.rawValue
+        dateComponents.second = self.second.nextValue()
+        dateComponents.minute = self.minute.nextValue()
+        dateComponents.hour = self.hour.nextValue()
+        dateComponents.weekday = self.day.nextValue()?.rawValue
+        dateComponents.day = self.date.nextValue()
+        dateComponents.month = self.month.nextValue()?.rawValue
+        return self.calendar.nextDate(after: date, matching: dateComponents, matchingPolicy: .strict)
+    }
+
+    ///  Set up scheduler to return the correct next date, based on a supplied Date.
+    /// - Parameter date: start date
+    public mutating func setInitialNextDate(after date: Date) -> Date? {
+        guard var nextDate = self.nextDate(after: date) else {
+            return nil
         }
-        if case .specific(let date) = self.date {
-            dateComponents.day = date
+        var prevDate = date
+        // Repeat while the nextDate is greater than the prevDate. At the point the nextDate is less than
+        // the previous date we know any schedules with multiple values have selected the correct next value
+        while prevDate < nextDate {
```

Comment:
nice! I think this will fix a current bug I notice where a job might not run until the metadata table is truncated. I have open a ticket yet because I am trying to get a reproducible example.